go/adbc/driver/bigquery/statement.go

```diff
@@ -0,0 +1,490 @@
+		}
+	case OptionStringQueryDefaultProjectID:
+		st.queryConfig.DefaultProjectID = v
+	case OptionStringQueryDefaultDatasetID:
+		st.queryConfig.DefaultDatasetID = v
+	case OptionStringQueryCreateDisposition:
+		val, err := stringToTableCreateDisposition(v)
+		if err == nil {
+			st.queryConfig.CreateDisposition = val
+		} else {
+			return true, err
+		}
+	case OptionStringQueryWriteDisposition:
+		val, err := stringToTableWriteDisposition(v)
+		if err == nil {
+			st.queryConfig.WriteDisposition = val
+		} else {
+			return true, err
+		}
+	case OptionBoolQueryDisableQueryCache:
+		val, err := stringToBool(v)
+		if err == nil {
+			st.queryConfig.DisableQueryCache = val
+		} else {
+			return true, err
+		}
+	case OptionBoolDisableFlattenedResults:
+		val, err := stringToBool(v)
+		if err == nil {
+			st.queryConfig.DisableFlattenedResults = val
+		} else {
+			return true, err
+		}
+	case OptionBoolQueryAllowLargeResults:
+		val, err := stringToBool(v)
+		if err == nil {
+			st.queryConfig.AllowLargeResults = val
+		} else {
+			return true, err
+		}
+	case OptionStringQueryPriority:
+		val, err := stringToQueryPriority(v)
+		if err == nil {
+			st.queryConfig.Priority = val
+		} else {
+			return true, err
+		}
+	case OptionBoolQueryUseLegacySQL:
+		val, err := stringToBool(v)
+		if err == nil {
+			st.queryConfig.UseLegacySQL = val
+		} else {
+			return true, err
+		}
+	case OptionBoolQueryDryRun:
+		val, err := stringToBool(v)
+		if err == nil {
+			st.queryConfig.DryRun = val
+		} else {
+			return true, err
+		}
+	case OptionBoolQueryCreateSession:
+		val, err := stringToBool(v)
+		if err == nil {
+			st.queryConfig.CreateSession = val
+		} else {
+			return true, err
+		}
+
+	default:
+		return false, nil
+	}
+	return true, nil
+}
+
+func (st *statement) setQueryConfigInt(key string, value int64) (bool, error) {
+	switch key {
+	case OptionIntQueryMaxBillingTier:
+		st.queryConfig.MaxBillingTier = int(value)
+	case OptionIntQueryMaxBytesBilled:
+		st.queryConfig.MaxBytesBilled = value
+	case OptionIntQueryJobTimeout:
+		st.queryConfig.JobTimeout = time.Duration(value) * time.Millisecond
+	default:
+		return false, nil
+	}
+	return true, nil
+}
+
+func (st *statement) SetOptions(options map[string]string) error {
+	for k, v := range options {
+		v := v // copy into loop scope
+		handled, err := st.setQueryConfig(k, v)
+		if handled {
+			if err != nil {
+				return err
+			} else {
+				continue
+			}
+		}
+
+		switch k {
+		default:
+			return adbc.Error{
+				Code: adbc.StatusInvalidArgument,
+				Msg:  fmt.Sprintf("unknown database string type option `%s`", k),
+			}
+		}
+	}
+	return nil
+}
+
+func (st *statement) SetOption(key string, value string) error {
+	handled, err := st.setQueryConfig(key, value)
+	if handled {
+		if err != nil {
+			return err
+		} else {
+			return nil
+		}
+	}
+
+	return adbc.Error{
+		Code: adbc.StatusInvalidArgument,
+		Msg:  fmt.Sprintf("unknown statement string type option `%s`", key),
+	}
+}
```

Comment:
make `setQueryConfig` return *only* the error, and then just turn this into:

`return st.setQueryConfig(key, value)`